src/routes/blog/post/introducing-python-machine-learning-runtime/+page.markdoc

```diff
@@ -0,0 +1,117 @@
+def normalize_img(image, label):
+    return tf.cast(image, tf.float32) / 255.0, label
+
+
+def main(context):
+    (ds_train, ds_test), ds_info = tfds.load(
+        "mnist",
+        split=["train", "test"],
+        shuffle_files=True,
+        as_supervised=True,
+        with_info=True,
+    )
+
+    ds_train = ds_train.map(normalize_img, num_parallel_calls=tf.data.AUTOTUNE)
+    ds_train = ds_train.cache()
+    ds_train = ds_train.shuffle(ds_info.splits["train"].num_examples)
+    ds_train = ds_train.batch(128)
+    ds_train = ds_train.prefetch(tf.data.AUTOTUNE)
+
+    ds_test = ds_test.map(normalize_img, num_parallel_calls=tf.data.AUTOTUNE)
+    ds_test = ds_test.batch(128)
+    ds_test = ds_test.cache()
+    ds_test = ds_test.prefetch(tf.data.AUTOTUNE)
+
+    model = tf.keras.models.Sequential(
+        [
+            tf.keras.layers.Flatten(input_shape=(28, 28)),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dense(10),
+        ]
+    )
+
+    model.compile(
+        optimizer=tf.keras.optimizers.Adam(0.001),
+        loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+        metrics=[tf.keras.metrics.SparseCategoricalAccuracy()],
+    )
+
+    model.fit(
+        ds_train,
+        epochs=6,
+        validation_data=ds_test,
+    )
+
+    return context.res.json({"success": True})
+```
+
+Find the full source code in our [GitHub repository](https://github.com/open-runtimes/open-runtimes/tree/main/runtimes/python-ml-3.11).
+
+## Best practices
+
+The Python ML runtime is in *beta.* We plan to iterate in response to developer demand and needs. For now, the runtime is that it is CPU-only. Here are some tips and tricks to help you overcome this limitation:
+
+### Smaller models & Tensorflow Lite
+
+Smaller models are faster to load and execute. Avoid using large models that require a lot of memory and CPU resources.
+
+Tensorflow Lite models are optimized for mobile and edge devices. They are smaller in size and can be run on CPUs. You can convert your Tensorflow models to Tensorflow Lite models using the [Tensorflow Lite Converter](https://www.tensorflow.org/lite/convert).
+
+### Async executions
+
+Synchronous executions have a global timeout limit of 30s on Appwrite Cloud. Therefore, it is recommended to use asynchronous function executions for long running tasks. This will allow you to run multiple tasks concurrently and make the most of the CPU resources available. Create an asynchronous execution using the `async` parameter in the [createExecution endpoint]
+(https://appwrite.io/docs/references/cloud/client-web/functions#createExecution)..
+
+## Training vs inference
+
+The Python ML runtime is optimized for inference tasks. Training models on the runtime is not recommended as it can be slow and resource-intensive. We recommend training your models on a local machine or a cloud-based GPU instance and then deploying the trained model to the Python ML runtime for inference.
```

Comment:
```suggestion
The Python ML runtime is more suitable for inference tasks. Training models on the runtime is not recommended as it can be slow and resource-intensive due to it being CPU only during the beta. We recommend training your models on a local machine or a cloud-based GPU instance and then deploying the trained model to the Python ML runtime for inference.
```